### PR TITLE
add modal for inscription and connexion

### PR DESCRIPTION
**Pull request by ozen2**

### CODE REVIEW

**Review by axeljdev — APPROVED** [review]

**Review by adrale — APPROVED** [review]

**Review by JulienChevreauPro — APPROVED** [review]

File: client/src/contexts/ModalContext.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      setIsOpen(!isOpen);
+    }, 1);
+  }, [isVisible, isOpen]);
+  window.scrollTo({
+    top: 0,
+    behavior: "smooth",
+  });
+
+  const isModalOpen = isOpen
+    ? "absolute opacity-100 delay-75 duration-500 top-0 bg-[var(--primary-background-color)] z-50 pb-16 flex flex-col items-center mt-6 border border-[var(--primary-color)] rounded-md shadow-lg custom-shadow mx-4 md:mx-0"
```
Comment (Yoann-Fortin) [inline client/src/contexts/ModalContext.jsx:34]: Too much duplicate code

File: client/src/components/Presentation.jsx [hunk 2]
```diff
@@ -1,15 +1,19 @@
 import Button from "./atomic/buttons/Button";
+import { useModal } from "../contexts/ModalContext";
 
 function Presentation() {
+  const {handleChangeModal} = useModal()
+
   return (
     <main className="flex flex-col items-center">
```
Comment (Yoann-Fortin) [inline client/src/components/Presentation.jsx:8]: How many main tag have you ?

File: client/src/components/Values.jsx
Comment (Yoann-Fortin) [inline client/src/components/Values.jsx:1]: Inappropriate use of section and article tags

File: client/src/components/ModalRegistration.jsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      </form>
+      <footer className="mt-10 mx-4 flex flex-col gap-10 items-center">
+        <CheckBox
+          id="CGU"
+          apply="bigCheckbox"
+          labelContent={
+            <>
+              J’ai lu et j’accepte les{" "}
+              <Link
+                to="/"
```
Comment (Yoann-Fortin) [inline client/src/components/ModalRegistration.jsx:92]: Redirect to real page

File: client/src/components/ModalRegistration.jsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+      return () => {
+        navAccessKeyboard.removeEventListener();
+      };
+    }
+    return () => {};
+  }, [modalRef]);
+
+  return (
+    <section
```
Comment (Yoann-Fortin) [inline client/src/components/ModalRegistration.jsx:28]: Dialog tag here please

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

**Review by Yoann-Fortin — APPROVED** [review]

**Pull request merged**